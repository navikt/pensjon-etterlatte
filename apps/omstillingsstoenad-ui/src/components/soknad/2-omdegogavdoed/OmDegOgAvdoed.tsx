import SoknadSteg from '../../../typer/SoknadSteg'
import { useTranslation } from 'react-i18next'
import { useSoknadContext } from '../../../context/soknad/SoknadContext'
import { ISoekerOgAvdoed } from '../../../typer/person'
import { ActionTypes } from '../../../context/soknad/soknad'
import { FormProvider, useForm } from 'react-hook-form'
import ForholdTilAvdoedeSkjema from './forholdTilAvdoede/ForholdTilAvdoedeSkjema'
import Feilmeldinger from '../../felles/Feilmeldinger'
import { Heading } from '@navikt/ds-react'
import Navigasjon from '../../felles/Navigasjon'
import { deepCopy } from '../../../utils/deepCopy'
import { SkjemaElement } from '../../felles/SkjemaElement'

const OmDegOgAvdoed: SoknadSteg = ({ neste, forrige }) => {
    const { t } = useTranslation()
    const { state, dispatch } = useSoknadContext()

    const methods = useForm<ISoekerOgAvdoed>({
        defaultValues: state.omDegOgAvdoed || {},
        shouldUnregister: true,
    })

    const {
        handleSubmit,
        formState: { errors },
        getValues,
    } = methods

    const erValidert = state.omDegOgAvdoed.erValidert

    const lagreNeste = (data: ISoekerOgAvdoed) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG_OG_AVDOED, payload: { ...deepCopy(data), erValidert: true } })
        neste!!()
    }

    const lagreTilbake = (data: ISoekerOgAvdoed) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG_OG_AVDOED, payload: { ...deepCopy(data), erValidert: true } })
        forrige!!()
    }

    const lagreTilbakeUtenValidering = () => {
        const verdier = getValues()
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG_OG_AVDOED, payload: { ...deepCopy(verdier), erValidert: false } })
        forrige!!()
    }

    return (
        <>
            <SkjemaElement>
                <Heading size={'medium'} className={'center'}>
                    {t('omDegOgAvdoed.tittel')}
                </Heading>
            </SkjemaElement>

            <FormProvider {...methods}>
                <form>
                    <ForholdTilAvdoedeSkjema />

                    <Feilmeldinger errors={errors} />

                    <Navigasjon
                        forrige={{
                            onClick: erValidert === true ? handleSubmit(lagreTilbake) : lagreTilbakeUtenValidering,
                        }}
                        neste={{ onClick: handleSubmit(lagreNeste) }}
                    />
                </form>
            </FormProvider>
        </>
    )
}

export default OmDegOgAvdoed
