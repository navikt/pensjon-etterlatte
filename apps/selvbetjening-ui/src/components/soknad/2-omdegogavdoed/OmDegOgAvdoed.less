@import "../../../App.less";

.opplysninger {
    .typo-element,
    .typo-normal {
        font-size: 1.1rem;
    }

    .typo-normal {
        margin: 0.2rem 0 1rem 0;
    }
}
