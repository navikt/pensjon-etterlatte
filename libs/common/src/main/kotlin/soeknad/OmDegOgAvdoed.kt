package soeknad

import com.fasterxml.jackson.annotation.JsonValue
import no.nav.etterlatte.libs.common.soeknad.Valg


data class OmDegOgAvdoed(
    val avdoed: Avdoed,
    val forholdTilAvdoede: ForholdTilAvdoede,
    val nySivilstatus: NySivilstatus
)

data class Avdoed(
    val datoForDoedsfallet: String,
    val etternavn: String,
    val fornavn: String
)

data class ForholdTilAvdoede(
    val relasjon: ForholdTilAvdoedeType,
    val datoForInngaattPartnerskap: String?,
    val datoForSkilsmisse: String?,
    val datoForInngaattSamboerskap: String?,
    val datoForSamlivsbrudd: String?,
    val fellesBarn: Valg?,
    val samboereMedFellesBarn: Valg?,
    val omsorgForBarn: Valg?,
    val tidligereGift: Valg?,
    val mottokBidrag: Valg?,
    val mottokEktefelleBidrag: Valg?
)

enum class ForholdTilAvdoedeType(@get:JsonValue val relasjon: String) {
    gift("avdoede.relasjon.gift"),
    separert("avdoede.relasjon.separert"),
    samboer("avdoede.relasjon.samboer"),
    skilt("avdoede.relasjon.skilt"),
    tidligereSamboer("avdoede.relasjon.tidligereSamboer")
}
