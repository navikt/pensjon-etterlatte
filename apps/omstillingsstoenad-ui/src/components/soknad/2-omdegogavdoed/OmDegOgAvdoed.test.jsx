import { act, render, fireEvent } from '@testing-library/react'
import OmDegOgAvdoed from './OmDegOgAvdoed'
import { describe, expect, it, vi } from 'vitest'
import {BrowserRouter} from "react-router-dom";

vi.mock('react-i18next', () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    ...vi.importActual('react-i18next'),
    useTranslation: () => ({
        t: (str) => str,
        i18n: {
            changeLanguage: () => new Promise(() => {}),
        },
    }),
}))

vi.mock('uuid', () => ({
    v4: () => '456',
}))

describe('Om deg og avdÃ¸d', () => {
    beforeAll(() => {
        vi.useFakeTimers('modern')
    })

    afterAll(() => {
        vi.useRealTimers()
    })

    it('Snapshot uten samboerskjema', () => {
        vi.setSystemTime(new Date(2024, 0, 1))
        const { container } = render(<BrowserRouter><OmDegOgAvdoed /></BrowserRouter>)
        expect(container).toMatchSnapshot()
    })

    it('Snapshot med sivilstatus samboer', () => {
        vi.setSystemTime(new Date(2024, 0, 1))
        const { container, getByText, getByLabelText } = render(<BrowserRouter><OmDegOgAvdoed /></BrowserRouter>)

        act(() => {
            fireEvent.click(getByLabelText('avdoede.relasjon.samboer'))
        })
        expect(getByText('omDegOgAvdoed.forholdTilAvdoede.fellesBarn')).toBeDefined()
        expect(container).toMatchSnapshot()
    })
})
