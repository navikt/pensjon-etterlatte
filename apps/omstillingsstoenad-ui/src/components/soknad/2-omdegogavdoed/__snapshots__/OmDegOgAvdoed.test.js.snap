// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Om deg og avd√∏d Snapshot med sivilstatus samboer 1`] = `
<div>
  <div
    class="sc-gueYoa qDReO"
  >
    <h1
      class="center navds-heading navds-heading--medium"
    >
      omDegOgAvdoed.tittel
    </h1>
  </div>
  <form>
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <label
        class="navds-label"
      >
        omDegOgAvdoed.avdoed.hvem
      </label>
      <div
        class="navds-grid"
      >
        <div
          class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
        >
          <div>
            <div
              class="navds-form-field navds-form-field--medium"
            >
              <label
                class="navds-form-field__label navds-label"
                for="avdoed.fornavn"
              >
                omDegOgAvdoed.avdoed.fornavn
              </label>
              <input
                aria-invalid="false"
                class="navds-text-field__input navds-body-short navds-body-medium"
                id="avdoed.fornavn"
                required=""
                type="text"
                value=""
              />
              <div
                aria-live="polite"
                aria-relevant="additions removals"
                class="navds-form-field__error"
                id="textField-error-rf"
              />
            </div>
          </div>
        </div>
        <div
          class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
        >
          <div>
            <div
              class="navds-form-field navds-form-field--medium"
            >
              <label
                class="navds-form-field__label navds-label"
                for="avdoed.etternavn"
              >
                omDegOgAvdoed.avdoed.etternavn
              </label>
              <input
                aria-invalid="false"
                class="navds-text-field__input navds-body-short navds-body-medium"
                id="avdoed.etternavn"
                required=""
                type="text"
                value=""
              />
              <div
                aria-live="polite"
                aria-relevant="additions removals"
                class="navds-form-field__error"
                id="textField-error-rg"
              />
            </div>
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <label
        class="navds-label"
      >
        omDegOgAvdoed.avdoed.datoForDoedsfallet
      </label>
      <section
        class="sc-hLseeU kvvimm"
      >
        <label
          class="navds-label"
          for="avdoed.datoForDoedsfallet"
        >
          omDegOgAvdoed.avdoed.dato felles.datoformat
        </label>
        <div
          class="datepicker"
        >
          <div>
            <div
              class="ds-datepicker"
            >
              <div
                class="navds-form-field navds-form-field--medium"
              >
                <label
                  class="navds-form-field__label navds-label"
                  for="avdoed.datoForDoedsfallet"
                />
                <div
                  style="position: relative;"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    autocorrect="off"
                    class="navds-text-field__input navds-body-short navds-body-medium"
                    id="avdoed.datoForDoedsfallet"
                    inputmode="text"
                    maxlength="10"
                    name="avdoed.datoForDoedsfallet"
                    placeholder="felles.datoEksempel"
                    type="text"
                    value=""
                  />
                  <button
                    aria-expanded="false"
                    class="ds-datepicker__calendarButton ds-datepicker__calendarButton--medium"
                    type="button"
                  >
                    <span
                      class="sr-only"
                    >
                      Kalender
                    </span>
                    <span
                      aria-hidden="true"
                      class="ds-datepicker__calendarButton__icon"
                    >
                      <svg
                        focusable="false"
                        height="16"
                        role="presentation"
                        style="pointer-events: none;"
                        viewBox="0 0 18 18"
                        width="16"
                      >
                        <title>
                          Kalenderikon
                        </title>
                        <g
                          fill="none"
                          fill-rule="evenodd"
                          stroke="#0067C5"
                        >
                          <path
                            d="M4 2.667H1.333v14h15.334v-14H14"
                          />
                          <path
                            d="M4 1.333h2V4H4zm8 0h2V4h-2zM6 2h6M1.333 6h15.334"
                          />
                        </g>
                      </svg>
                    </span>
                  </button>
                </div>
                <div
                  aria-live="polite"
                  aria-relevant="additions removals"
                  class="navds-form-field__error"
                  id="textField-error-rh"
                />
              </div>
            </div>
          </div>
        </div>
      </section>
    </fieldset>
    <div
      id="forholdTilAvdoede.relasjon"
    >
      <div
        class="sc-gueYoa qDReO"
      >
        <fieldset
          aria-invalid="false"
          class="navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
        >
          <legend
            class="navds-fieldset__legend navds-label"
          >
            omDegOgAvdoed.forholdTilAvdoede.relasjon
          </legend>
          <div
            class="navds-radio-buttons"
          >
            <div
              class="radioBorder navds-radio navds-radio--medium"
            >
              <input
                aria-invalid="false"
                class="navds-radio__input"
                id="radio-rk"
                name="forholdTilAvdoede.relasjon"
                type="radio"
                value="avdoede.relasjon.gift"
              />
              <label
                class="navds-radio__label"
                for="radio-rk"
              >
                <span
                  class="navds-radio__content"
                >
                  <span
                    class="navds-body-short"
                  >
                    avdoede.relasjon.gift
                  </span>
                </span>
              </label>
            </div>
            <div
              class="radioBorder navds-radio navds-radio--medium"
            >
              <input
                aria-invalid="false"
                class="navds-radio__input"
                id="radio-rl"
                name="forholdTilAvdoede.relasjon"
                type="radio"
                value="avdoede.relasjon.separert"
              />
              <label
                class="navds-radio__label"
                for="radio-rl"
              >
                <span
                  class="navds-radio__content"
                >
                  <span
                    class="navds-body-short"
                  >
                    avdoede.relasjon.separert
                  </span>
                </span>
              </label>
            </div>
            <div
              class="radioBorder navds-radio navds-radio--medium"
            >
              <input
                aria-invalid="false"
                class="navds-radio__input"
                id="radio-rm"
                name="forholdTilAvdoede.relasjon"
                type="radio"
                value="avdoede.relasjon.samboer"
              />
              <label
                class="navds-radio__label"
                for="radio-rm"
              >
                <span
                  class="navds-radio__content"
                >
                  <span
                    class="navds-body-short"
                  >
                    avdoede.relasjon.samboer
                  </span>
                </span>
              </label>
            </div>
            <div
              class="radioBorder navds-radio navds-radio--medium"
            >
              <input
                aria-invalid="false"
                class="navds-radio__input"
                id="radio-rn"
                name="forholdTilAvdoede.relasjon"
                type="radio"
                value="avdoede.relasjon.skilt"
              />
              <label
                class="navds-radio__label"
                for="radio-rn"
              >
                <span
                  class="navds-radio__content"
                >
                  <span
                    class="navds-body-short"
                  >
                    avdoede.relasjon.skilt
                  </span>
                </span>
              </label>
            </div>
            <div
              class="radioBorder navds-radio navds-radio--medium"
            >
              <input
                aria-invalid="false"
                class="navds-radio__input"
                id="radio-ro"
                name="forholdTilAvdoede.relasjon"
                type="radio"
                value="avdoede.relasjon.tidligereSamboer"
              />
              <label
                class="navds-radio__label"
                for="radio-ro"
              >
                <span
                  class="navds-radio__content"
                >
                  <span
                    class="navds-body-short"
                  >
                    avdoede.relasjon.tidligereSamboer
                  </span>
                </span>
              </label>
            </div>
          </div>
          <div
            aria-live="polite"
            aria-relevant="additions removals"
            class="navds-fieldset__error"
            id="fieldset-error-rj"
          />
        </fieldset>
      </div>
    </div>
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <div
        class="sc-gueYoa qDReO"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          omDegOgAvdoed.nySivilstatus.sivilstatus
        </h1>
        <p
          class="navds-body-long"
        >
          omDegOgAvdoed.nySivilstatus.beskrivelse
        </p>
      </div>
      <div
        id="nySivilstatus.sivilstatus"
      >
        <div
          class="sc-gueYoa qDReO"
        >
          <fieldset
            aria-invalid="false"
            class="navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
          >
            <legend
              class="navds-fieldset__legend navds-label"
            />
            <div
              class="navds-radio-buttons"
            >
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  aria-invalid="false"
                  class="navds-radio__input"
                  id="radio-rr"
                  name="nySivilstatus.sivilstatus"
                  type="radio"
                  value="nySivilstatus.enslig"
                />
                <label
                  class="navds-radio__label"
                  for="radio-rr"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short"
                    >
                      nySivilstatus.enslig
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  aria-invalid="false"
                  class="navds-radio__input"
                  id="radio-rs"
                  name="nySivilstatus.sivilstatus"
                  type="radio"
                  value="nySivilstatus.ekteskap"
                />
                <label
                  class="navds-radio__label"
                  for="radio-rs"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short"
                    >
                      nySivilstatus.ekteskap
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  aria-invalid="false"
                  class="navds-radio__input"
                  id="radio-rt"
                  name="nySivilstatus.sivilstatus"
                  type="radio"
                  value="nySivilstatus.samboerskap"
                />
                <label
                  class="navds-radio__label"
                  for="radio-rt"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short"
                    >
                      nySivilstatus.samboerskap
                    </span>
                  </span>
                </label>
              </div>
            </div>
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-fieldset__error"
              id="fieldset-error-rq"
            />
          </fieldset>
        </div>
      </div>
      <div
        class="navds-panel navds-panel--border"
      >
        <div
          class="sc-gueYoa qDReO"
        >
          <h1
            class="navds-heading navds-heading--small"
          >
            omDegOgAvdoed.nySivilstatus.samboerskap.samboer.tittel
          </h1>
        </div>
        <div
          class="sc-gueYoa qDReO"
        >
          <div
            class="navds-grid"
          >
            <div
              class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
            >
              <div
                class="kol-50"
              >
                <div
                  class="navds-form-field navds-form-field--medium"
                >
                  <label
                    class="navds-form-field__label navds-label"
                    for="nySivilstatus.samboerskap.samboer.fornavn"
                  >
                    omDegOgAvdoed.nySivilstatus.samboerskap.samboer.fornavn
                  </label>
                  <input
                    aria-invalid="false"
                    class="navds-text-field__input navds-body-short navds-body-medium"
                    id="nySivilstatus.samboerskap.samboer.fornavn"
                    required=""
                    type="text"
                    value=""
                  />
                  <div
                    aria-live="polite"
                    aria-relevant="additions removals"
                    class="navds-form-field__error"
                    id="textField-error-ru"
                  />
                </div>
              </div>
            </div>
            <div
              class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
            >
              <div
                class="kol-50"
              >
                <div
                  class="navds-form-field navds-form-field--medium"
                >
                  <label
                    class="navds-form-field__label navds-label"
                    for="nySivilstatus.samboerskap.samboer.etternavn"
                  >
                    omDegOgAvdoed.nySivilstatus.samboerskap.samboer.etternavn
                  </label>
                  <input
                    aria-invalid="false"
                    class="navds-text-field__input navds-body-short navds-body-medium"
                    id="nySivilstatus.samboerskap.samboer.etternavn"
                    required=""
                    type="text"
                    value=""
                  />
                  <div
                    aria-live="polite"
                    aria-relevant="additions removals"
                    class="navds-form-field__error"
                    id="textField-error-rv"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-gueYoa qDReO"
        >
          <div
            class="navds-form-field navds-form-field--medium"
          >
            <label
              class="navds-form-field__label navds-label"
              for="nySivilstatus.samboerskap.samboer.foedselsnummer"
            >
              felles.fnr
            </label>
            <input
              aria-invalid="false"
              class="navds-text-field__input navds-body-short navds-body-medium"
              id="nySivilstatus.samboerskap.samboer.foedselsnummer"
              placeholder="felles.fnrPlaceholder"
              required=""
              size="25"
              type="tel"
              value=""
            />
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-form-field__error"
              id="textField-error-r10"
            />
          </div>
        </div>
        <div
          class="sc-gueYoa qDReO"
        >
          <div
            id="nySivilstatus.samboerskap.hattBarnEllerVaertGift"
          >
            <fieldset
              aria-invalid="false"
              class="sc-dmqHEX hKqYBB navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
            >
              <legend
                class="navds-fieldset__legend navds-label"
              >
                omDegOgAvdoed.nySivilstatus.samboerskap.hattBarnEllerVaertGift
              </legend>
              <div
                class="navds-radio-buttons"
              >
                <div
                  class="radioBorder navds-radio navds-radio--medium"
                >
                  <input
                    aria-invalid="false"
                    class="navds-radio__input"
                    id="radio-r13"
                    name="nySivilstatus.samboerskap.hattBarnEllerVaertGift"
                    type="radio"
                    value="radiobuttons.ja"
                  />
                  <label
                    class="navds-radio__label"
                    for="radio-r13"
                  >
                    <span
                      class="navds-radio__content"
                    >
                      <span
                        class="navds-body-short"
                      >
                        radiobuttons.ja
                      </span>
                    </span>
                  </label>
                </div>
                <div
                  class="radioBorder navds-radio navds-radio--medium"
                >
                  <input
                    aria-invalid="false"
                    class="navds-radio__input"
                    id="radio-r14"
                    name="nySivilstatus.samboerskap.hattBarnEllerVaertGift"
                    type="radio"
                    value="radiobuttons.nei"
                  />
                  <label
                    class="navds-radio__label"
                    for="radio-r14"
                  >
                    <span
                      class="navds-radio__content"
                    >
                      <span
                        class="navds-body-short"
                      >
                        radiobuttons.nei
                      </span>
                    </span>
                  </label>
                </div>
              </div>
              <div
                aria-live="polite"
                aria-relevant="additions removals"
                class="navds-fieldset__error"
                id="fieldset-error-r12"
              />
            </fieldset>
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <div
        class="sc-eDDNvR kRBBlR undefined"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            aria-live="polite"
            class="navds-label"
          >
            knapp.tilbake
          </span>
        </button>
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            aria-live="polite"
            class="navds-label"
          >
            knapp.neste
          </span>
        </button>
      </div>
    </fieldset>
    <div
      class="sc-eDDNvR sc-jTrPJq kRBBlR konaIR"
    >
      <button
        class="navds-button navds-button--secondary navds-button--medium"
        id="avbryt-btn"
        type="button"
      >
        <span
          aria-live="polite"
          class="navds-label"
        >
          knapp.avbryt
        </span>
      </button>
    </div>
  </form>
</div>
`;

exports[`Om deg og avd√∏d Snapshot uten samboerskjema 1`] = `
<div>
  <div
    class="sc-gueYoa qDReO"
  >
    <h1
      class="center navds-heading navds-heading--medium"
    >
      omDegOgAvdoed.tittel
    </h1>
  </div>
  <form>
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <label
        class="navds-label"
      >
        omDegOgAvdoed.avdoed.hvem
      </label>
      <div
        class="navds-grid"
      >
        <div
          class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
        >
          <div>
            <div
              class="navds-form-field navds-form-field--medium"
            >
              <label
                class="navds-form-field__label navds-label"
                for="avdoed.fornavn"
              >
                omDegOgAvdoed.avdoed.fornavn
              </label>
              <input
                aria-invalid="false"
                class="navds-text-field__input navds-body-short navds-body-medium"
                id="avdoed.fornavn"
                required=""
                type="text"
                value=""
              />
              <div
                aria-live="polite"
                aria-relevant="additions removals"
                class="navds-form-field__error"
                id="textField-error-r0"
              />
            </div>
          </div>
        </div>
        <div
          class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6"
        >
          <div>
            <div
              class="navds-form-field navds-form-field--medium"
            >
              <label
                class="navds-form-field__label navds-label"
                for="avdoed.etternavn"
              >
                omDegOgAvdoed.avdoed.etternavn
              </label>
              <input
                aria-invalid="false"
                class="navds-text-field__input navds-body-short navds-body-medium"
                id="avdoed.etternavn"
                required=""
                type="text"
                value=""
              />
              <div
                aria-live="polite"
                aria-relevant="additions removals"
                class="navds-form-field__error"
                id="textField-error-r1"
              />
            </div>
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <label
        class="navds-label"
      >
        omDegOgAvdoed.avdoed.datoForDoedsfallet
      </label>
      <section
        class="sc-hLseeU kvvimm"
      >
        <label
          class="navds-label"
          for="avdoed.datoForDoedsfallet"
        >
          omDegOgAvdoed.avdoed.dato felles.datoformat
        </label>
        <div
          class="datepicker"
        >
          <div>
            <div
              class="ds-datepicker"
            >
              <div
                class="navds-form-field navds-form-field--medium"
              >
                <label
                  class="navds-form-field__label navds-label"
                  for="avdoed.datoForDoedsfallet"
                />
                <div
                  style="position: relative;"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    autocorrect="off"
                    class="navds-text-field__input navds-body-short navds-body-medium"
                    id="avdoed.datoForDoedsfallet"
                    inputmode="text"
                    maxlength="10"
                    name="avdoed.datoForDoedsfallet"
                    placeholder="felles.datoEksempel"
                    type="text"
                    value=""
                  />
                  <button
                    aria-expanded="false"
                    class="ds-datepicker__calendarButton ds-datepicker__calendarButton--medium"
                    type="button"
                  >
                    <span
                      class="sr-only"
                    >
                      Kalender
                    </span>
                    <span
                      aria-hidden="true"
                      class="ds-datepicker__calendarButton__icon"
                    >
                      <svg
                        focusable="false"
                        height="16"
                        role="presentation"
                        style="pointer-events: none;"
                        viewBox="0 0 18 18"
                        width="16"
                      >
                        <title>
                          Kalenderikon
                        </title>
                        <g
                          fill="none"
                          fill-rule="evenodd"
                          stroke="#0067C5"
                        >
                          <path
                            d="M4 2.667H1.333v14h15.334v-14H14"
                          />
                          <path
                            d="M4 1.333h2V4H4zm8 0h2V4h-2zM6 2h6M1.333 6h15.334"
                          />
                        </g>
                      </svg>
                    </span>
                  </button>
                </div>
                <div
                  aria-live="polite"
                  aria-relevant="additions removals"
                  class="navds-form-field__error"
                  id="textField-error-r2"
                />
              </div>
            </div>
          </div>
        </div>
      </section>
    </fieldset>
    <div
      id="forholdTilAvdoede.relasjon"
    >
      <div
        class="sc-gueYoa qDReO"
      >
        <fieldset
          aria-invalid="false"
          class="navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
        >
          <legend
            class="navds-fieldset__legend navds-label"
          >
            omDegOgAvdoed.forholdTilAvdoede.relasjon
          </legend>
          <div
            class="navds-radio-buttons"
          >
            <div
              class="radioBorder navds-radio navds-radio--medium"
            >
              <input
                aria-invalid="false"
                class="navds-radio__input"
                id="radio-r5"
                name="forholdTilAvdoede.relasjon"
                type="radio"
                value="avdoede.relasjon.gift"
              />
              <label
                class="navds-radio__label"
                for="radio-r5"
              >
                <span
                  class="navds-radio__content"
                >
                  <span
                    class="navds-body-short"
                  >
                    avdoede.relasjon.gift
                  </span>
                </span>
              </label>
            </div>
            <div
              class="radioBorder navds-radio navds-radio--medium"
            >
              <input
                aria-invalid="false"
                class="navds-radio__input"
                id="radio-r6"
                name="forholdTilAvdoede.relasjon"
                type="radio"
                value="avdoede.relasjon.separert"
              />
              <label
                class="navds-radio__label"
                for="radio-r6"
              >
                <span
                  class="navds-radio__content"
                >
                  <span
                    class="navds-body-short"
                  >
                    avdoede.relasjon.separert
                  </span>
                </span>
              </label>
            </div>
            <div
              class="radioBorder navds-radio navds-radio--medium"
            >
              <input
                aria-invalid="false"
                class="navds-radio__input"
                id="radio-r7"
                name="forholdTilAvdoede.relasjon"
                type="radio"
                value="avdoede.relasjon.samboer"
              />
              <label
                class="navds-radio__label"
                for="radio-r7"
              >
                <span
                  class="navds-radio__content"
                >
                  <span
                    class="navds-body-short"
                  >
                    avdoede.relasjon.samboer
                  </span>
                </span>
              </label>
            </div>
            <div
              class="radioBorder navds-radio navds-radio--medium"
            >
              <input
                aria-invalid="false"
                class="navds-radio__input"
                id="radio-r8"
                name="forholdTilAvdoede.relasjon"
                type="radio"
                value="avdoede.relasjon.skilt"
              />
              <label
                class="navds-radio__label"
                for="radio-r8"
              >
                <span
                  class="navds-radio__content"
                >
                  <span
                    class="navds-body-short"
                  >
                    avdoede.relasjon.skilt
                  </span>
                </span>
              </label>
            </div>
            <div
              class="radioBorder navds-radio navds-radio--medium"
            >
              <input
                aria-invalid="false"
                class="navds-radio__input"
                id="radio-r9"
                name="forholdTilAvdoede.relasjon"
                type="radio"
                value="avdoede.relasjon.tidligereSamboer"
              />
              <label
                class="navds-radio__label"
                for="radio-r9"
              >
                <span
                  class="navds-radio__content"
                >
                  <span
                    class="navds-body-short"
                  >
                    avdoede.relasjon.tidligereSamboer
                  </span>
                </span>
              </label>
            </div>
          </div>
          <div
            aria-live="polite"
            aria-relevant="additions removals"
            class="navds-fieldset__error"
            id="fieldset-error-r4"
          />
        </fieldset>
      </div>
    </div>
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <div
        class="sc-gueYoa qDReO"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          omDegOgAvdoed.nySivilstatus.sivilstatus
        </h1>
        <p
          class="navds-body-long"
        >
          omDegOgAvdoed.nySivilstatus.beskrivelse
        </p>
      </div>
      <div
        id="nySivilstatus.sivilstatus"
      >
        <div
          class="sc-gueYoa qDReO"
        >
          <fieldset
            aria-invalid="false"
            class="navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
          >
            <legend
              class="navds-fieldset__legend navds-label"
            />
            <div
              class="navds-radio-buttons"
            >
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  aria-invalid="false"
                  class="navds-radio__input"
                  id="radio-rc"
                  name="nySivilstatus.sivilstatus"
                  type="radio"
                  value="nySivilstatus.enslig"
                />
                <label
                  class="navds-radio__label"
                  for="radio-rc"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short"
                    >
                      nySivilstatus.enslig
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  aria-invalid="false"
                  class="navds-radio__input"
                  id="radio-rd"
                  name="nySivilstatus.sivilstatus"
                  type="radio"
                  value="nySivilstatus.ekteskap"
                />
                <label
                  class="navds-radio__label"
                  for="radio-rd"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short"
                    >
                      nySivilstatus.ekteskap
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  aria-invalid="false"
                  class="navds-radio__input"
                  id="radio-re"
                  name="nySivilstatus.sivilstatus"
                  type="radio"
                  value="nySivilstatus.samboerskap"
                />
                <label
                  class="navds-radio__label"
                  for="radio-re"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short"
                    >
                      nySivilstatus.samboerskap
                    </span>
                  </span>
                </label>
              </div>
            </div>
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-fieldset__error"
              id="fieldset-error-rb"
            />
          </fieldset>
        </div>
      </div>
    </fieldset>
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <div
        class="sc-eDDNvR kRBBlR undefined"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            aria-live="polite"
            class="navds-label"
          >
            knapp.tilbake
          </span>
        </button>
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            aria-live="polite"
            class="navds-label"
          >
            knapp.neste
          </span>
        </button>
      </div>
    </fieldset>
    <div
      class="sc-eDDNvR sc-jTrPJq kRBBlR konaIR"
    >
      <button
        class="navds-button navds-button--secondary navds-button--medium"
        id="avbryt-btn"
        type="button"
      >
        <span
          aria-live="polite"
          class="navds-label"
        >
          knapp.avbryt
        </span>
      </button>
    </div>
  </form>
</div>
`;
